api/models/dogs.js

```diff
@@ -0,0 +1,64 @@
+var chunk = require('lodash/chunk');
+var redis = require('../database');
+
+
+exports.save = function(dog) {
+  if (dog && dog.btle_devices && dog.btle_devices.length > 0) {
+    var deviceId = dog.btle_devices[0].device_id;
+    redis.sadd('dogs', dog.display_id);
+    redis.sadd('btle_devices', deviceId);
+    redis.set('dog:' + dog.display_id, JSON.stringify(dog));
+    redis.set('btle_device:' + deviceId + ':dog', dog.display_id);
+  }
+};
+
+exports.getAllDogs = function(callback) {
+  redis.smembers('dogs', function(err, dogs) {
+    var multi = redis.multi();
+    dogs.forEach(function(dogId) {
+      multi.get('dog:' + dogId);
+      multi.get('dog:' + dogId + ':checked_in');
+    });
+
+    multi.exec(function(err, replies) {
+      var dogs = chunk(replies, 2).map(function(d) {
+        var dog = JSON.parse(d[0]);
+        var checkin = JSON.parse(d[1]);
+
+        return {
+          display_id: dog.display_id,
+          name: dog.name,
+          short_description: dog.short_description,
+          owner: dog.owner,
+          avatar: dog.avatar,
+          checked_in: checkin
+        };
+      });
+      callback(dogs);
+    });
+  });
+};
+
+exports.getDogById = function(dogId, callback) {
+  var multi = redis.multi();
+  multi.get('dog:' + dogId);
+  multi.get('dog:' + dogId + ':checked_in');
+
+  multi.exec(function(err, d) {
+    var dog = JSON.parse(d[0]);
+    var checkin = JSON.parse(d[1]);
+
+    if (dog) {
+      callback({
```

Comment:
The convention in node is to reserve the first argument for any errors you want to propagate and send data as the second param.